CMakeLists.txt

```diff
@@ -195,6 +195,9 @@ endif()
 if(ROBOTOLOGY_ENABLE_ICUB_HEAD)
   find_or_build_package(icub-firmware-shared)
   find_or_build_package(ICUB)
+  if(NOT APPLE)
```

Comment:
Can we at least give it a shot to compile it under `macOS` ? Otherwise, we need to explicitly document that `diagnosticdaemon` is not compile in macOS in the README, but if we can avoid to do this it would be great.